client/cli/src/lib.rs

```diff
@@ -197,10 +197,15 @@ pub trait SubstrateCli: Sized {
 		command: &T,
 	) -> error::Result<Runner<Self>> {
 		let tokio_runtime = build_runtime()?;
+
+		// `capture` needs to be called in a tokio context.
+		// Also capture them as early as possible.
+		let signals = tokio_runtime.block_on(async { Signals::capture() })?;
```

Comment:
(Maybe dumb questions) Why `capture` needs to be called in a tokio context? Why not just: `let signals = Signals::capture()?;`

Reply:
This is a requirement of most of the Tokio stuff. When you call something in a Tokio context, it will set some thread local variable to access the context. They need this to register certain things.